dispatch/dispatch.go

```diff
@@ -0,0 +1,295 @@
+var MaxWorkers int64
+
+// comms is our main instance
+var comms *Communications
+
+// init initial startup (✿◠‿◠)
+func init() {
+	comms = &Communications{
+		Routing: make(map[uuid.UUID][]chan interface{}),
+		jobs:    make(chan *job, defaultJobBuffer),
+		outbound: sync.Pool{
+			New: func() interface{} {
+				// Create unbuffered channel for data pass
+				return make(chan interface{})
+			},
+		},
+	}
+
+	MaxWorkers = DefaultMaxWorkers
+	// TODO: Might drop this worker in the future and just allocate and
+	// de-allocate as workers are needed
+	go comms.relayer()
+}
+
+// job defines a relaying job associated with a ticket which allows routing to
+// routines that require specific data
+type job struct {
+	Data interface{}
+	ID   uuid.UUID
+	Err  chan interface{}
+}
+
+// Communications defines inner-subsystem communication systems
+type Communications struct {
+	// Outbound subystem with a ticket association so this could be routes will
+	// be cleaned when sub-systems have no use with them
+	// TODO: limit slice doubling
+	Routing map[uuid.UUID][]chan interface{}
+	rwMtx   sync.RWMutex
+
+	// Persistent job channel see job struct
+	jobs chan *job
+
+	// Dynamic channel communication pools; unbuffered outbound channels for
+	// generic data and buffered inbound channels for general errors
+	outbound sync.Pool
+	inbound  sync.Pool
+
+	// Atomic worker count
+	count int64
+	// Atomic worker gateway
+	gateway uint32
+}
+
+// Relayer routine relays communications across the defined routes
+func (c *Communications) relayer() {
+	atomic.AddInt64(&c.count, 1)
+	tick := time.NewTicker(defaultGatewaySleep)
+	for {
+		select {
+		case j := <-c.jobs:
+			c.rwMtx.RLock()
+			if _, ok := c.Routing[j.ID]; !ok {
+				c.rwMtx.RUnlock()
+				continue
+			}
+			// Channel handshake timeout feature if a channel is blocked for any
+			// period of time due to an issue with the receiving routine.
+			// This will wait on channel then fall over to the next route when
+			// the timer actuates and continue over the route list. Have to
+			// iterate across full length of routes so every routine can get
+			// their new info, cannot be buffered as we dont want to have an old
+			// orderbook contained in a buffered channel when a routine actually
+			// is ready for a receive.
+			// TODO: Need to consider optimal timer length
+			chanHSTimeout := time.NewTimer(handshakeTimeout)
+			for i := range c.Routing[j.ID] {
+				select {
+				case c.Routing[j.ID][i] <- j.Data:
+				case <-chanHSTimeout.C:
+				}
+				// Reset timeout when data is put in the last channel route or
+				// if it fails in this individual route
+				chanHSTimeout.Reset(handshakeTimeout)
+			}
+			c.rwMtx.RUnlock()
+
+		case <-tick.C:
+			if atomic.CompareAndSwapUint32(&c.gateway, 0, 1) {
+				if len(c.jobs) < (defaultJobBuffer / defaultWorkerRate) {
+					if atomic.LoadInt64(&c.count) > 1 {
+						atomic.AddInt64(&c.count, -1)
+						atomic.SwapUint32(&c.gateway, 0)
+						return
```

Comment:
Will do

Reply:
calling a stop on the tick and timer before returning doesn't close any of its underlying channels or increase effectiveness of garbage collection, its sole purpose is to stop the firing of that channel. Still a go routine can read from this without issues it will just block or continue if the buffered time channel is loaded. 

This *should be* garbage collected effectively because there are no out of scope references to any other routine.

Reply:
I know it doesn't close the channel, but I'm going from tick documentation:
```
// Stop the ticker to release associated resources.
func NewTicker(d Duration) *Ticker {
```

Reply:
No you are right, my mistake it has some lovely low level run time code that needs to be freed.